src/test/java/com/google/sps/PlaceGuideServletTest.java

```diff
@@ -0,0 +1,615 @@
+  private static final String EMAIL_USER_C = "emailUserC";
+  private static final String NAME_USER_C = "nameUserC";
+  private static final boolean PUBLIC_PORTFOLIO_USER_C = true;
+  private static final String SELF_INTRODUCTION_USER_C = "selfIntroductionUserC";
+  private static final String IMG_KEY_USER_C = "imgKeyUserC";
+
+  private final User userC =
+      new User.Builder(ID_USER_C, EMAIL_USER_C)
+          .setBookmarkedPlaceGuidesIds(Collections.emptySet())
+          .setName(NAME_USER_C)
+          .setPublicPortfolio(true)
+          .addSelfIntroduction(SELF_INTRODUCTION_USER_C)
+          .addImgKey(IMG_KEY_USER_C)
+          .build();
+
+  // Creator D data.
+  private static final String ID_USER_D = "idUserD";
+  private static final String EMAIL_USER_D = "emailUserD";
+  private static final String NAME_USER_D = "nameUserD";
+  private static final boolean PUBLIC_PORTFOLIO_USER_D = true;
+  private static final String SELF_INTRODUCTION_USER_D = "selfIntroductionUserD";
+  private static final String IMG_KEY_USER_D = "imgKeyUserD";
+
+  private final User userD =
+      new User.Builder(ID_USER_D, EMAIL_USER_D)
+          .setBookmarkedPlaceGuidesIds(Collections.emptySet())
+          .setName(NAME_USER_D)
+          .setPublicPortfolio(true)
+          .addSelfIntroduction(SELF_INTRODUCTION_USER_D)
+          .addImgKey(IMG_KEY_USER_D)
+          .build();
+
+  // General placeguide data(used for all placeguides).
+  private static final String NAME = "name";
+  private static final String AUDIO_KEY = "audioKey";
+  private static final String PLACE_ID = "placeId";
+  private static final boolean IS_PUBLIC = true;
+  private static final long LENGTH = new Long(60);
+  private static final String DESCRIPTION = "description";
+  private static final String IMAGE_KEY = "imageKey";
+
+  // PlaceGuides' parameters used for map-related queries.
+  // PlaceGuides of user C.
+  private static final long C_INNER_PUBLIC_ID = 56789;
+  private static final long C_INNER_PRIVATE_ID = 98765;
+  private static final long C_OUTER_PUBLIC_ID = 67890;
+  private static final long C_OUTER_PRIVATE_ID = 9876;
+  private static final String CREATOR_C_ID = ID_USER_C;
+  private static final GeoPt C_INNER_PUBLIC_COORDINATE = new GeoPt((float) 10, (float) -5);
+  private static final GeoPt C_INNER_PRIVATE_COORDINATE = new GeoPt((float) -14, (float) 14);
+  private static final GeoPt C_OUTER_PUBLIC_COORDINATE = new GeoPt((float) 5, (float) -20);
+  private static final GeoPt C_OUTER_PRIVATE_COORDINATE = new GeoPt((float) -30, (float) -5);
+  // PlaceGuides of user D.
+  private static final long D_INNER_PUBLIC_ID = 567890;
+  private static final long D_INNER_PRIVATE_ID = 987650;
+  private static final long D_OUTER_PUBLIC_ID = 678900;
+  private static final long D_OUTER_PRIVATE_ID = 98760;
+  private static final String CREATOR_D_ID = ID_USER_D;
+  private static final GeoPt D_INNER_PUBLIC_COORDINATE = new GeoPt((float) 10, (float) 5);
```

Comment:
`10f` instead of `(float) 10` here and below

Reply:
Done.